tests/test_utils.py

```diff
@@ -35,3 +62,50 @@ def test_memory_human_readable_conversion():
     assert bytes_to_human_readable(1024) == (1.0, "KB")
     assert bytes_to_human_readable(1024**2) == (1.0, "MB")
     assert bytes_to_human_readable(4.3 * 1024**4) == (4.3, "TB")
+
+
+@pytest.mark.parametrize(
+    ("target, hook, error"),
+    [
+        param(
+            f_annotated,
+            f_annotated_one_default,
+            None,
+            id="hook_with_defaults",
+        ),
+        param(
+            f_annotated_one_default,
+            f_annotated,
+            None,
+            id="target_with_defaults",
+        ),
+        param(
+            f_annotated,
+            f_plain,
+            None,
+            id="hook_without_annotations",
+        ),
+        param(
+            f_annotated,
+            f_reversed_annotated,
+            TypeError,
+            id="different_order",
+        ),
+        param(
+            f_annotated,
+            f2_plain,
+            TypeError,
+            id="different_names",
+        ),
+        param(
+            f_annotated,
+            f_reduced_plain,
+            TypeError,
+            id="hook_missing_arguments",
+        ),
+    ],
+)
+def test_register_hook(target, hook, error):
+    """Passing in-/consistent signatures to `register_hook` raises an/no error."""
+    with pytest.raises(error) if error is not None else nullcontext():
```

Comment:
i think design is cleaner if we split valid and invalid tests into two separate functions

Reply:
Ok, if you prefer. We have that pattern also in other places, but I changed it now